### PR TITLE
feat(protocol-designer): Default tc state for to previous saved state

### LINKED ISSUE #5577: TC Default State Form to Last State
As a user, I would like my TC step for to default to the TC’s last saved state

## Acceptance Criteria
- [ ] When user opens a new state form, default all fields to current TC state.

## Implementation detail

This is a follow-up expanding on `createPresavedStepForm` functionality initiated in #5596

### PR DESCRIPTION
**Pull request by Kadee80**
##  overview

This PR closes #5577 by adding getters for the defaults for previously saved TC state fields

## changelog

- feat(protocol-designer): Default tc state for to previous saved state
- added/updated tests

## review requests

Code stuff:
- [ ] Test updates are correct
- [ ] New tests cover everything

Make a protocol with multiple TC state steps
- [ ] Block Active/Deactivated populates with previously saved value
- [ ] Block Target Temp populates with previously saved value
- [ ] Lid Active/Deactivated populates with previously saved value
- [ ] LidTarget Temp populates with previously saved value
- [ ] Lid Open/Closed populates with previously saved value

## risk assessment

Low, PD behind a FF

### CODE REVIEW

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultBlockTemperature.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        )
+
+        expect(result).toBe(expected)
+      })
+    })
+  })
+
+  describe('with previous forms', () => {
+    const testCases = [
+      {
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultBlockTemperature.test.js:30]: these test case names are sorta hard to read, im not entirely sure what the expected behavior is. 

The first one seems clear enough: `returns null when block deactivated and no previous block temp entered`

The second one has an active, followed by deactive, followed by active. So it returns the the block temp from the previous step when the previous step is active and has a block temperature?

The third has deactive, followed by active, followed by deactive. So it returns the block temp from the last active step with a defined block temp? If so, does that contradict this AC: `Block Target Temp populates with previously saved value`? Since in this case, it's not actually the previously saved value, it could be the value from multiple steps ago.

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultBlockIsActive.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// @flow
+import { getNextDefaultBlockIsActive } from '../'
+
+describe('getNextDefaultBlockIsActive', () => {
+  describe('no previous forms defaults to false', () => {
+    const testCases = [
+      {
+        testMsg: 'no previous thermocycler state',
+        expected: false,
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultBlockIsActive.test.js:9]: can this be renamed to`returns false when no previous thermocycler state` so it follows a `does X when Y` pattern?

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultBlockTemperature.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+// @flow
+import { getNextDefaultBlockTemperature } from '../'
+
+describe('getNextDefaultBlockTemperature', () => {
+  describe('no previous forms defaults to null', () => {
+    const testCases = [
+      {
+        testMsg: 'no previous thermocycler state',
+        expected: null,
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultBlockTemperature.test.js:9]: same comment as above re:naming

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidIsActive.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// @flow
+import { getNextDefaultLidIsActive } from '../'
+
+describe('getNextDefaultLidIsActive', () => {
+  describe('no previous forms defaults to false', () => {
+    const testCases = [
+      {
+        testMsg: 'no previous thermocycler state',
+        expected: false,
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidIsActive.test.js:9]: same comment re:naming

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidOpen.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+// @flow
+import { getNextDefaultLidOpen } from '../'
+
+describe('getNextDefaultLidOpen', () => {
+  describe('no previous forms defaults to null', () => {
+    const testCases = [
+      {
+        testMsg: 'no previous thermocycler state',
+        expected: null,
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidOpen.test.js:9]: same comment re:naming

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidTemperature.test.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+// @flow
+import { getNextDefaultLidTemperature } from '../'
+
+describe('getNextDefaultLidTemperature', () => {
+  describe('no previous forms defaults to null', () => {
+    const testCases = [
+      {
+        testMsg: 'no previous thermocycler state',
+        expected: null,
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidTemperature.test.js:9]: same comment re:naming

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidTemperature.test.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        testMsg:
+          'returns null when lid deactivated and no previous lid temp entered',
+        orderedStepIds: ['d'],
+        expected: null,
+      },
+      {
+        testMsg:
+          'returns default when lid activated previous selected and previous lid temp entered',
+        orderedStepIds: ['a', 'd', 'a'],
+        expected: 40,
```
Comment (shlokamin) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidTemperature.test.js:37]: see my comment in `getNextDefaultBlockTemperature.test.js`, these names are a bit hard to follow

**Review by shlokamin — COMMENTED** [review]
Looks good and coverage seems pretty air tight! Left some comments about test naming and a question about one of the ACs

**Review by shlokamin — APPROVED** [review]
🚢

File: protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidOpen.test.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  describe('with previous forms', () => {
+    const testCases = [
+      {
+        testMsg: 'returns true when true previously selected',
+        orderedStepIds: ['t', 'f', 't'],
+        expected: true,
+      },
+      {
+        testMsg: 'returns false when false previously selected',
+        orderedStepIds: ['f', 'f', 'f'],
```
Comment (IanLondon) [inline protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidOpen.test.js:34]: is this meant to be `f t f`?

Comment (Kadee80) [reply protocol-designer/src/steplist/formLevel/getNextDefaultThermocyclerState/__tests__/getNextDefaultLidOpen.test.js:34]: yeah. good catch

**Review by IanLondon — APPROVED** [review]
code looks good, tested via UI also good :t-rex:
